tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/ESEmailAddressContactSearchEngine.java

```diff
@@ -42,14 +52,27 @@ public ESEmailAddressContactSearchEngine(ReactorElasticSearchClient client) {
 
     @Override
     public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String part) {
-        Mono.fromRunnable(() -> {
-            SearchRequest request = new SearchRequest("posts");
-            SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
-            QueryBuilder matchQueryBuilder = QueryBuilders.boolQuery()
-                    .filter(QueryBuilders.matchQuery("accountId", accountId.getIdentifier()))
-                    .should(QueryBuilders.matchQuery("address", "part"));
-            sourceBuilder.query(matchQueryBuilder);
-            request.source(sourceBuilder);
-        }).then();
+        SearchRequest request = new SearchRequest("posts");
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        QueryBuilder matchQueryBuilder = QueryBuilders.boolQuery()
+                .filter(QueryBuilders.matchQuery("accountId", accountId.getIdentifier()))
+                .should(QueryBuilders.matchQuery("address", "part"));
+        sourceBuilder.query(matchQueryBuilder);
+        request.source(sourceBuilder);
+        // A search request needs a client to invoke `search` method, a client could get in Test, which means
+        // i need a client as parameters to pass to this method, which will change the interface then it seems
+        // not quite right.
+        Mono<SearchResponse> searchResponseMono = client.search(request, RequestOptions.DEFAULT);
+        SearchResponse response = searchResponseMono.block();
```

Comment:
I don't think you should block here... You should have all your following logic in the Mono as a single pipeline, avoiding that block()

Reply:
Yes you are right, looking for a way to avoid that block. May you give some suggestion please?